controller/bindings/connman/connman.ml

```diff
@@ -270,10 +276,17 @@ struct
   ; strength : int option
   ; favorite : bool
   ; autoconnect : bool
+
+  (* The ipv4 field represents the actual system configuration,
+     while the ipv4_user_config represents values manually configured by the user.
+  *)
   ; ipv4 : IPv4.t option
+  ; ipv4_user_config : IPv4.t option
```

Comment:
I am wondering if we could merge `ipv4_user_config` with `ipv4` with the following logic: `ipv4_user_config.or_else(ipv4)`.

### Not configured:

```ocaml
(ipv4 
  (((method' dhcp) (address 192.168.1.1) (netmask 255.255.255.0) (gateway (192.168.1.200)))))
(ipv4_user_config 
  ())
```

### Configured:

```ocaml
(ipv4
  (((method' manual) (address 192.168.1.100) (netmask 255.255.255.0) (gateway (192.168.1.200)))))
(ipv4_user_config
  (((method' manual) (address 192.168.1.100) (netmask 255.255.255.0) (gateway (192.168.1.200)))))
```

Reply:
Good idea!